### PR TITLE
Fix the infinite loading state

### PR DESCRIPTION
**Pull request by siddharthisrani**
Description
This pull request addresses the issue of the infinite loading state.

Changes Made
Added error handling to set loading to false and clear viewData when an error occurs.
Updated the "p" tag to display "No data available" with professional styling using Tailwind CSS.

Checklist
 Tested the changes locally.
 Code adheres to project standards.
Thank you!

Fix the infinite loading state.
nodatafound

### CODE REVIEW

File: src/components/PageViews/ArticleViewsGraph.jsx [hunk 1]
```diff
@@ -61,7 +62,13 @@ const ArticleViewsGraph = ({ articles }) => {
...
         <div className='graph w-full flex flex-col items-center'>
             <h2 className='font-bold text-white text-xl self-start'>Views for Multiple Articles</h2>
-            {loading ? <Loading /> : viewData.labels ? <Line data={viewData} /> : <p>No data available.</p>}
+            {loading ? (
+                <Loading />
+            ) : viewData.labels ? (
+                <Line data={viewData} />
+            ) : (
+                <p className='text-white text-2xl absolute top-1/2 right-1/4 transform -translate-x-1/2 -translate-y-1/2'>No data available.</p>
+            )}
```
Comment (CREDO23) [inline src/components/PageViews/ArticleViewsGraph.jsx:71]: It is a good start @siddharthisrani !
It seems to work the first time, but not for subsequent tries.

Try the following :

I have noticed some issues with the `<ArticleViewsGraph/>` component:

1. Set the default value of loading to false :
```typescript
    const [loading, setLoading] = useState(false);
```

2. Then set it to true when a new API request is made 

```typescript
     useEffect(() => {
        const fetchAllViews = async () => {
            try {

               setLoading(true);  // <------

                const allDataPromises = articles.map(async (article) => {
                    const response = await fetchPageViewsCount(article);
                    const views = response.items;
                    const dates = views.map((view) => {
                        const year = view.timestamp.substring(0, 4);
                        const month = view.timestamp.substring(4, 6);
                        const day = view.timestamp.substring(6, 8);
                        const dateString = `${year}-${month}-${day}`;
                        const date = new Date(dateString);
                        return date.toLocaleDateString();
                    });
                    const counts = views.map((view) => view.views);
                    return { article: article.article, dates, counts };
                });
                .........
```
3. Add a `finally` block in which you will set the loading state to false : (Whether the request succeeds or failed, we need to make sure the loading is set to false so we can handle the error properly)

4. In the `catch` block : Set data to `{}`

**Review by CREDO23 — CHANGES_REQUESTED** [review]
Hi @siddharthisrani, we appreciate your contribution !

However, there are few changes to add before merging your PR. Please see the comments !

File: src/components/PageViews/ArticleViewsGraph.jsx (hunk 1 above)
Comment (siddharthisrani) [reply src/components/PageViews/ArticleViewsGraph.jsx:71]: Hii @CREDO23 ,
Sure, I'll make those changes and update the PR accordingly. Thank you for the feedback!

I'll work on setting the default value of loading to false, updating it to true when a new API request is made, and adding a finally block to set loading back to false. I'll also handle setting the data to an empty object in the catch block.

I'll let you know once the changes are done.

Comment (CREDO23) [reply src/components/PageViews/ArticleViewsGraph.jsx:71]: Perfect @siddharthisrani !

**Comment by siddharthisrani** [comment]
Hi @CREDO23,

I have updated the PR with the requested changes. Here are the modifications made to the ArticleViewsGraph.jsx component:

1. Set the default value of loading to false.
2. Set loading to true when a new API request is made.
3. Added a finally block to set loading to false.
4. Cleared viewData in the catch block if an error occurs.
These changes should ensure the component behaves correctly on subsequent tries by managing the loading state properly and handling errors effectively.

Please review the updated code and let me know if any further changes are needed.

**Comment by CREDO23** [comment]
Have you checked the feature out ? In local ?

**Comment by siddharthisrani** [comment]
Yes, I have tested the updated feature locally. The loading state management and error handling in the ArticleViewsGraph.jsx component are working as expected.

Please let me know if there are any other changes or tests you would like me to perform.

**Comment by CREDO23** [comment]
Great 👍, let me check!

**Comment by siddharthisrani** [comment]
Thanks, looking forward to your feedback!

**Comment by CREDO23** [comment]
Looks good to me :+1: ! @siddharthisrani

**Pull request merged**